thrust/system/detail/generic/shuffle.inl

```diff
@@ -99,10 +99,11 @@ class feistel_bijection {
   // from random for each key value input. This is not cryptographically secure
   // but sufficient for generating permutations. 
   __host__ __device__ uint32_t round_function(uint64_t value,
-                                              const uint64_t key) const {
-    uint64_t hash0 = thrust::random::taus88(value)();
+                                              const uint64_t key_) const {
+    uint64_t hash0 = thrust::random::taus88(static_cast<uint32_t>(value))();
```

Comment:
Can this be `std::uint32_t` and `std::uint64_t`?

Reply:
No -- `tau88` only accepts `uint32_t` inputs. This doesn't change anything, it just makes the implicit cast explicit and silences the conversion warnings.

 I checked with the `shuffle` author about this change and he was fine with it, it shouldn't affect the results.

Reply:
I just meant stick `std::` in front of the types.

Reply:
Done.